internal/allocator/allocator.go

```diff
@@ -14,6 +14,16 @@ import (
 	"github.com/mikioh/ipaddr"
 )
 
+// IP family helps identifying single stack IPv4/IPv6 vs Dual-stack ["IPv4", "IPv6"] or ["IPv6", "Ipv4"].
+type IPFamily int
+
+const (
+	IPv4 IPFamily = iota
```

Comment:
I'd use a string (still defined as a custom type) so it's easier to understand when we dump / log.

```
type Family string

const (
	IPV4      Family = "IPV4"
	IPV6      Family = "IPV6"
	DualStack Family = "DualStack"
)
```